cartographer_ros/cartographer_ros/offline_node.cc

```diff
@@ -233,12 +237,24 @@ void RunOfflineNode(const MapBuilderFactory& map_builder_factory) {
 
   // TODO(gaschler): Warn if resolved topics are not in bags.
   std::unordered_map<int, int> bag_index_to_trajectory_id;
+  cartographer::common::optional<ros::Time> begin_time;
   while (playable_bag_multiplexer.IsMessageAvailable()) {
+    if (!::ros::ok()) {
+      return;
+    }
+
     const auto next_msg_tuple = playable_bag_multiplexer.GetNextMessage();
     const rosbag::MessageInstance& msg = std::get<0>(next_msg_tuple);
     const int bag_index = std::get<1>(next_msg_tuple);
     const bool is_last_message_in_bag = std::get<2>(next_msg_tuple);
 
+    if (!begin_time.has_value()) {
```

Comment:
Is there no way to peek the first message rather than having to do this check every time?

Reply:
We can extend `PlayableBagMultiplexer` to allow for peeking the time of the next message, if you prefer.

Reply:
@ojura Awesome! Let's do it!